ecosystem/typescript/sdk/src/plugins/ans_client.ts

```diff
@@ -132,6 +143,111 @@ export class AnsClient {
     return pendingTransaction.hash;
   }
 
+  /**
+   * Mint a new Aptos Subdomain
+   *
+   * @param account AptosAccount the owner of the domain name
+   * @param subdomainName subdomain name to mint
+   * @param domainName Aptos domain name to mint under
+   * @param expirationTimestamp must be set between the domains expiration and the current time
+   * @returns The hash of the pending transaction submitted to the API
+   */
+  async mintAptosSubdomain(
+    account: AptosAccount,
+    subdomainName: string,
+    domainName: string,
+    expirationTimestampSeconds?: number,
+    extraArgs?: OptionalTransactionArgs,
+  ): Promise<Gen.HashValue> {
+    // check if the name is valid
+    if (domainName.match(nameComponentPattern) === null) {
+      throw new ApiError(400, `Domain name ${domainName} is not valid`);
+    }
+    // check if the name is valid
+    if (subdomainName.match(nameComponentPattern) === null) {
+      throw new ApiError(400, `Subdomain name ${subdomainName} is not valid`);
+    }
+    // check if the name is available
+    const subdomainRegistration = await this.getRegistrationForSubdomainName(domainName, subdomainName);
+    if (subdomainRegistration !== null) {
+      const now = Math.ceil(Date.now() / 1000);
+      if (now < subdomainRegistration.expirationTimestampSeconds) {
+        throw new ApiError(400, `Name ${subdomainName}.${domainName} is not available`);
+      }
+    }
+
+    const domainRegistration = await this.getRegistrationForDomainName(domainName);
+    if (domainRegistration === null) {
+      throw new ApiError(400, `Domain name ${domainName} does not exist`);
+    }
+    const now = Math.ceil(Date.now() / 1000);
+    if (domainRegistration.expirationTimestampSeconds < now) {
```

Comment:
could we add some explanation for all the `expirationTimestampSeconds` check we do?

Reply:
Probably could make a function that is: `registration.isExpired(now)` rather than a comment, but I feel like it's fairly straightforward given the error below it.